src/core/mac/mac.cpp

```diff
@@ -1154,6 +1156,11 @@ ThreadError Mac::ProcessReceiveSecurity(Frame &aFrame, const Address &aSrcAddr,
             mKeyManager.SetCurrentKeySequence(keySequence);
         }
     }
+    else if (keyIdMode == Frame::kKeyIdMode2)
+    {
+        // reset frame counter
+        mKeyIdMode2FrameCounter = 0;
```

Comment:
My reading of Section 7.2.1.3 indicates that an MLE Announce message should never be rejected because of the Frame Counter value.  We should file a DEV JIRA if that is the case.